cmd/up/activate.go

```diff
@@ -457,3 +462,29 @@ func getPullingMessage(message, namespace string) string {
 	toReplace := fmt.Sprintf("%s/%s", registry, namespace)
 	return strings.Replace(message, toReplace, okteto.DevRegistry, 1)
 }
+
+// waitUntilAppIsAwaken waits until the app is awaken checking if the annotation dev.okteto.com/state-before-sleeping is present in the app resource
+func (up *upContext) waitUntilAppIsAwaken(ctx context.Context, app apps.App) error {
+	if _, ok := app.ObjectMeta().Annotations[model.StateBeforeSleepingAnnontation]; !ok {
+		return nil
+	}
+
+	timeout := 5 * time.Minute
+	to := time.NewTicker(timeout)
+	ticker := time.NewTicker(10 * time.Second)
+	for {
+		select {
+		case <-to.C:
+			return fmt.Errorf("'%s' '%s' didn't wake up after %s", app.Kind(), app.ObjectMeta().Name, timeout.String())
+		case <-ticker.C:
+			if err := app.Refresh(ctx, up.Client); err != nil {
+				return err
+			}
+
+			// If the app is not sleeping anymore, we are done
+			if _, ok := app.ObjectMeta().Annotations[model.StateBeforeSleepingAnnontation]; !ok {
+				return nil
+			}
```

Comment:
Probably it is a good idea to print something here to the user in case we have to wait too much, WDYT?

Reply:
yes IMO. User might think that there is something wrong when launching his dev env and he just has to be a little patient.

Reply:
Yes, I'll include a message

Reply:
Done! I included a spinner